doc/build-netbsd.md

```diff
@@ -1,92 +1,144 @@
+
+```bash
+pkgin install db4
+```
+
+###### Descriptor Wallet Support
+
+`sqlite3` is required to enable support for [descriptor wallets](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md). Skip if you don't intend to use descriptor wallets.
 
-The descriptor wallet uses `sqlite3`. You can install it using:
 ```bash
 pkgin install sqlite3
 ```
 
+#### GUI Dependencies
+
+Bitcoin Core includes a GUI built with the cross-platform Qt Framework. To compile the GUI, we need to install `qt5`. Skip if you don't intend to use the GUI.
+
 ```bash
-./autogen.sh
-./configure --with-gui=no --without-bdb \
-    CPPFLAGS="-I/usr/pkg/include" \
-    LDFLAGS="-L/usr/pkg/lib" \
-    BOOST_CPPFLAGS="-I/usr/pkg/include" \
-    MAKE=gmake
+pkgin install qt5
 ```
 
-#### With legacy wallet:
+The GUI can encode addresses in a QR Code. To build in QR support for the GUI, install `qrencode`. Skip if not using the GUI or don't need QR code functionality.
+
+```bash
+pkgin install qrencode
+```
 
-BerkeleyDB is use for legacy wallet functionality.
+#### Test Suite Dependencies
 
-It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
-from ports.
-You can use [the installation script included in contrib/](/contrib/install_db4.sh) like so:
+There is an included test suite that is useful for testing code changes when developing.
+To run the test suite (recommended), you will need to have Python 3 installed:
 
 ```bash
-./contrib/install_db4.sh `pwd`
+pkgin install python37
 ```
 
-from the root of the repository. Then set `BDB_PREFIX` for the next section:
+### Building Bitcoin Core
+
+**Note**: Use `gmake` (the non-GNU `make` will exit with an error).
+
+
+### 1. Configuration
+
+There are many ways to configure Bitcoin Core. Here are a few common examples:
+##### Wallet (BDB + SQlite) Support, No GUI:
+
+This explicitly enables legacy wallet support and disables the GUI. An error will be thrown if `db4` is not installed. If `sqlite3` is installed, then descriptor wallet support will also be built.
 
 ```bash
-export BDB_PREFIX="$PWD/db4"
+./autogen.sh
+./configure --with-gui=no \
+    CPPFLAGS="-I/usr/pkg/include" \
+    LDFLAGS="-L/usr/pkg/lib" \
+    --with-boost-libdir=/usr/pkg/lib \
+    MAKE=gmake
 ```
 
+##### Wallet (only SQlite) and GUI Support:
+
+This explicitly enables the GUI and disables legacy wallet support. If `qt5` is not installed, this will throw an error. If `sqlite3` is installed then descriptor wallet functionality will be built. If `sqlite3` is not installed, then wallet functionality will be disabled.
+
 ```bash
 ./autogen.sh
-./configure --with-gui=no CPPFLAGS="-I/usr/pkg/include" \
+./configure --without-bdb --with-gui=yes \
+    CPPFLAGS="-I/usr/pkg/include" \
     LDFLAGS="-L/usr/pkg/lib" \
-    BOOST_CPPFLAGS="-I/usr/pkg/include" \
-    BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
-    BDB_CFLAGS="-I${BDB_PREFIX}/include" \
+    --with-boost-libdir=/usr/pkg/lib \
```

Comment:
Is using BOOST_CPPFLAGS not sufficient? We no-longer link against any Boost libraries.

Reply:
actually i confirmed the only thing we now need is CPPFLAGS